deepreg/dataset/preprocess.py

```diff
@@ -326,3 +324,158 @@ def resize_inputs(
         fixed_label=fixed_label,
         indices=indices,
     )
+
+
+def gen_rand_affine_transform(
+    batch_size: int, scale: float, seed: (int, None) = None
+) -> tf.Tensor:
+    """
+    Function that generates a random 3D transformation parameters for a batch of data.
+
+    for 3D coordinates, affine transformation is
+
+    .. code-block:: text
+
+        [[x' y' z' 1]] = [[x y z 1]] * [[* * * 0]
+                                        [* * * 0]
+                                        [* * * 0]
+                                        [* * * 1]]
+
+    where each * represents a degree of freedom,
+    so there are in total 12 degrees of freedom
+    the equation can be denoted as
+
+        new = old * T
```

Comment:
left versus right multiplication?

Reply:
emmm I was just moving docs, but this equation is consistent with L339 above? it depends on row or column vectors